include/swift/SILOptimizer/Utils/ConstExpr.h

```diff
@@ -0,0 +1,86 @@
+//===--- ConstExpr.h - Constant expression evaluator -----------*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// This defines an interface to evaluate Swift language level constant
+// expressions.  Its model is intended to be general and reasonably powerful,
+// with the goal of standardization in a future version of Swift.
+//
+// Constant expressions are functions without side effects that take constant
+// values and return constant values.  These constants may be integer, floating
+// point, and string values, or arrays thereof (up to 1024 elements).  We allow
+// abstractions to be built out of fragile structs and tuples.
+//
```

Comment:
@marcrasi  

> These constants may be integer, floating point, and string values, or arrays thereof (up to 1024 elements).

Should we remove this comment Line as we don't have array/string in this commit? Is the limit of 1024 elements imposed somewhere in the array extensions? If so, should we really commit to this constant?